### PR TITLE
Extensive modifications to overview.md

### PR DESCRIPTION
**Pull request by guydebros**
I tried to add clarity and details while maintaining the spirit of the original
document. However, it might be a bit too 'dry' now. I'd be happy to liven it up
a bit. I'm only serious when I feel like I _have_ to be.

One of my opinions (which may be at odds with yours) is that the documentation
should emphasize `jrnl`'s advantages without downplaying any other existing
solutions.

If I have time, I'd like to add more information about the documented benefits
of journaling, particularly the mental health aspects. That will probably need
its own page, but I'm new here, so I don't want to overstay my welcome. :)

### Checklist

- [ ] The code change is tested and works locally.
- [ ] Tests pass. Your PR cannot be merged unless tests pass. --
  `poetry run behave`
- [ ] The code passes linting via
  [black](https://black.readthedocs.io/en/stable/) (consistent code styling). --
  `poetry run black --check . --verbose --diff`
- [ ] The code passes linting via [pyflakes](https://launchpad.net/pyflakes)
  (logically errors and unused imports). -- `poetry run pyflakes jrnl features`
- [ ] There is no commented out code in this PR.
- [ ] Have you followed the guidelines in our Contributing document?
- [ ] Have you checked to ensure there aren't other open
  [Pull Requests](../pulls) for the same update/change?
- [ ] Have you added an explanation of what your changes do and why you'd like
  us to include them?
- [ ] Have you written new tests for your core changes, as applicable?

### CODE REVIEW

File: docs/overview.md [hunk 1]
```diff
@@ -1,27 +1,44 @@
...
+to improve mental health and quality of life.
 
-For professional use, consider a text-based journal to be the perfect
-complement to your GTD todo list - a documentation of what and how
-you've done it. Or use it as a quick way to keep a change log. Or use it
-to keep a lab book.
+For professional use, a text-based journal can be the perfect complement to your
+to-do list -- a record of what you've done as well as how and when you've done
+it. You can use a journal as a quick way to keep a change log (`jrnl Updated
+jrnl overview page to make it more awesomer`). Or use it to keep a lab notebook.
```
Comment (eshrh) [inline docs/overview.md:43]: comma after notebook, join the two fragments?

File: docs/overview.md [hunk 2]
```diff
@@ -1,27 +1,44 @@
...
+anything else that is worth remembering. Your imagination may be limitless, but
+your memory probably isn't.
 
-For personal use, make it a good habit to write at least 20 words a day.
-Just to reflect what made this day special, why you haven't wasted it.
+To begin, try to write at least 20 words a day. `jrnl` makes this incredibly
+easy. Fire up a terminal and type, `jrnl today at 8am: I arrived at work to find
+a birthday present on my desk. My colleagues are the best!` You can reflect on
+what made your day special so that you'll always remember why you haven't wasted
+it. Or try using `jrnl` to keep a gratitude journal, a habit that has been shown
```
Comment (eshrh) [inline docs/overview.md:37]: TBH I feel like the explanation for why this is good isn't extremely necessary. However if you're going to use the phrasing "has been shown to" I think people might expect a reference or a link or something.

File: docs/overview.md [hunk 3]
```diff
@@ -1,27 +1,44 @@
...
+Journals aren't just for people who have too much time on their summer vacation.
+Journals help you keep track of events, thoughts, feelings, tasks, ideas, and
+anything else that is worth remembering. Your imagination may be limitless, but
+your memory probably isn't.
 
-For personal use, make it a good habit to write at least 20 words a day.
-Just to reflect what made this day special, why you haven't wasted it.
+To begin, try to write at least 20 words a day. `jrnl` makes this incredibly
+easy. Fire up a terminal and type, `jrnl today at 8am: I arrived at work to find
+a birthday present on my desk. My colleagues are the best!` You can reflect on
```
Comment (eshrh) [inline docs/overview.md:35]: Perhaps blockquote or place the code on a separate line for fluent readability

File: docs/overview.md [hunk 4]
```diff
@@ -1,27 +1,44 @@
...
+  still be readable in 2050 and beyond. Entries can be searched, filtered, and
+  edited rapidly and easily.
+- Journals can be stored anywhere, including in shared folders so that they can
+  be synchronized between devices.
+- Journal entries can be created and modified using an external editor of your
+  choice, using the command line as a starting point to filter specific entries.
+- BeJournals are stored in single files using date/time tags to identify
+  individual entries. Some systems endorse a file/folder methodology; `jrnl`
+  keeps 
+- Journals can be protected using [256-bit
```
Comment (eshrh) [inline docs/overview.md:23]: This whole bullet point sequence seems rather repetitive. Every line starts with journals. Maybe it's possible to rephrase some differently?

File: docs/overview.md [hunk 5]
```diff
@@ -1,27 +1,44 @@
...
+- Journal entries are created and displayed directly from the command line.
+- Journals are stored as human-readable, future-proof plain text files. Because
+  files are stored as plain text, you can rest assured that your journals will
+  still be readable in 2050 and beyond. Entries can be searched, filtered, and
+  edited rapidly and easily.
+- Journals can be stored anywhere, including in shared folders so that they can
+  be synchronized between devices.
+- Journal entries can be created and modified using an external editor of your
+  choice, using the command line as a starting point to filter specific entries.
+- BeJournals are stored in single files using date/time tags to identify
```
Comment (eshrh) [inline docs/overview.md:20]: Typo.

File: docs/overview.md [hunk 6]
```diff
@@ -1,27 +1,44 @@
...
 
-## What is jrnl?
+## What is `jrnl`?
 
-`jrnl` is a simple journal application for
-your command line. Journals are stored as human readable plain text
-files - you can put them into a Dropbox folder for instant syncing and
-you can be assured that your journal will still be readable in 2050,
-when all your fancy iPad journal applications will long be forgotten.
+`jrnl` is a simple but powerful plain text journal application for the command
```
[the remark this answers is not in the document]
Comment (guydebros) [reply docs/overview.md:5]: At the least, it needs to be on its own line.

[the remark this answers is not in the document]
Comment (eshrh) [reply docs/overview.md:5]: I forgot to mention here, `plain text journal application` is repeated twice in close proximity. Perhaps that line could be something like

> `jrnl` is a simple but powerful plain text journal application for the command line.
> Although it's not the only one, it offers several features:

File: docs/overview.md (hunk 2 above)
Comment (guydebros) [reply docs/overview.md:37]: I agree that it isn't really necessary. I'm guessing someone looking for the docs has probably already made the decision to keep a journal and doesn't need any convincing.

File: docs/overview.md (hunk 5 above)
Comment (guydebros) [reply docs/overview.md:20]: ::facepalm::

File: docs/overview.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (guydebros) [reply docs/overview.md:3]: Yeah, that was ugly. I will fix it.

File: docs/overview.md (hunk 3 above)
Comment (guydebros) [reply docs/overview.md]: done

File: docs/overview.md [hunk 7]
```diff
@@ -1,44 +1,41 @@
...
-
-For professional use, a text-based journal can be the perfect complement to your
-to-do list -- a record of what you've done as well as how and when you've done
-it. You can use a journal as a quick way to keep a change log (`jrnl Updated
-jrnl overview page to make it more awesomer`). Or use it to keep a lab notebook.
-The possibilities are almost endless.
+Your imagination may be limitless, but your memory probably isn't. `jrnl` lets
+you write something down, and then it gets out of the way.
+
+## how it works
```
Comment (eshrh) [inline docs/overview.md:9]: I think these should be capital, but this is obviously just an aesthetic choice.

File: docs/overview.md [hunk 8]
```diff
@@ -1,44 +1,41 @@
...
+text, you can rest assured that your journals will be readable for centuries.
+
+### ...and as many files as you need
+  
+`jrnl` allows you to work with multiple journals, each of which is stored as a
+single file using date and time tags to identify individual entries. `jrnl`
+makes it easy to find the entries you want, and only the ones you want, so that
+you can read them or edit them. Here's an example: say you want to find all of
+the entries you wrote on January 3rd, 2020 that include the word _cat_, and
+change every instance of the word _cat_ to _dog_? Easy:
```
Comment (eshrh) [inline docs/overview.md:33]: This doesn't actually change anything, it ONLY gets entries with _cat_, so the text above it is inaccurate.

File: docs/overview.md (hunk 7 above)
Comment (guydebros) [reply docs/overview.md:9]: Like, 'How It Works,' or 'HOW IT WORKS'? The first way is the most traditional.

File: docs/overview.md (hunk 8 above)
Comment (guydebros) [reply docs/overview.md:33]: Right, I should clarify, you still have to do the find-and-replace-all in the editor, but at least it only shows the ones that matter. I'll think of a better example.

File: docs/overview.md (hunk 7 above)
Comment (eshrh) [reply docs/overview.md:9]: First one imo

Comment (MinchinWeb) [reply docs/overview.md:9]: All headings should probably be fixed. Look up "titlecase" if you want a deep dive into what words to capitalize and which ones not to.

File: docs/overview.md (hunk 8 above)
Comment (wren) [reply docs/overview.md:33]: I think removing the part about `_dog_` here would be enough to clarify the example.

File: docs/overview.md [hunk 9]
```diff
@@ -1,27 +1,41 @@
...
 
-For personal use, make it a good habit to write at least 20 words a day.
-Just to reflect what made this day special, why you haven't wasted it.
+Support for external editors is included.
 
-For professional use, consider a text-based journal to be the perfect
-complement to your GTD todo list - a documentation of what and how
-you've done it. Or use it as a quick way to keep a change log. Or use it
-to keep a lab book.
+### one file type to rule them all...
```
Comment (wren) [inline docs/overview.md:19]: As mentioned above, the titles here should all be in title case.

File: docs/overview.md [hunk 10]
```diff
@@ -1,27 +1,41 @@
...
+
+`jrnl` stores your journals as human-readable, future-proof plain text files.
+You can store them wherever you want, including in shared folders to keep them
+synchronized between devices. And because journal files are stored as plain
+text, you can rest assured that your journals will be readable for centuries.
+
+### ...and as many files as you need
+  
+`jrnl` allows you to work with multiple journals, each of which is stored as a
+single file using date and time tags to identify individual entries. `jrnl`
```
Comment (wren) [inline docs/overview.md:29]: This isn't true anymore, so maybe it needs a bit of clarification. You can have as many journals as you want, and each journal can either be stored in a single large file, or as multiple files (sorted by date) in a directory (directory-based, multiple-file journals are a newer feature we implemented recently).

Comment (guydebros) [reply docs/overview.md:29]: Cool! I definitely missed that. I'll update it to include the new options.

File: docs/overview.md (hunk 9 above)
Comment (guydebros) [reply docs/overview.md:19]: - Can you clarify what you mean about the `_dog_` thing? Delete it entirely, maybe? It was intended to emphasize ease of use, but it is kind of superfluous. Might be better on the advanced usage page.

- planning on changing titles to title case, and ditching the LOTR reference (unless people like it)

**Pull request merged**